actions/event-parsers.js

```diff
@@ -82,25 +104,24 @@ module.exports = {
   tpch: (body, v) => {
     let $ = cheerio.load(body);
     let events = [];
+    const month = moment().format('MMMM');
 
-    $('#blog li').first().find('p').each((i,e) => {
-      let date;
-      let artist;
+    $(`h1:contains(${month})`).parent('header').parent('article').find('p').each((i,e) => {
       const listing = $(e).text();
-      const listingArr = listing.split('~');
+      const [ day, artist ]= listing.split('~');
+      let date = moment(`${month} ${day}`, 'MMMM ddd Do');
 
-      if (listingArr.length > 1) {
-        date = listingArr[0].trim();
-        date += ' 7pm (Sunday 1pm)';
-        artist = listingArr[1].trim();
-        events.push({ date, artist });
-      }
-      else if (listingArr.length === 1) {
-        const tmpVar = listingArr[0].trim();
-        if (tmpVar != '') {
-          date = artist = tmpVar;
-          events.push({ date, artist });
-        }
+      if (date.isValid() && dateData.isThisWeek(date)) {
+        const dayName = date.format('dddd');
+        const eventTime = dayName === 'Sunday' ? '1:00 PM' : '7:00 PM'
```

Comment:
What would be better though is still persisting that time to the data object. Keep the data consistent and perhaps avoid any odd side effects down the road for some stupid reason.